File: BackEnd_ExpressJS/app/models/itemMenu.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const Translation = require('./translation');

const ItemMenu = sequelize.define('tblItemMenu', {
        ideItemMenu: {
            type: DataTypes.BIGINT,
            primaryKey: true,
        },
        strLink: {
            type: DataTypes.TEXT,
            allowNull: false,
        },
        ideTranslation: {
            type: DataTypes.BIGINT,
            allowNull: true,
            references: {
              model: Translation,
              key: 'ideTranslation',
            },
            onUpdate: 'CASCADE',
            onDelete: 'SET NULL',
        },
        createdAt: {
            type: DataTypes.DATE,
            allowNull: false,
        },
        updatedAt: {
            type: DataTypes.DATE,
            allowNull: false,
        },
    }, {
        // Activar/Desactivar las marcas de tiempo
        timestamps: true,
        // Usar el nombre de la tabla tal como está en la base de datos
        freezeTableName: true,
    });

ItemMenu.belongsTo(Translation, { foreignKey: 'ideTranslation' });

module.exports = ItemMenu;

File: BackEnd_Django/api/myModels/itemMenu.py
from django.db import models
from .translation import Translation

class ItemMenu(models.Model):
    ideItemMenu = models.BigAutoField(primary_key=True)
    strLink = models.TextField()
    ideTranslation = models.ForeignKey(Translation, on_delete=models.SET_NULL, null=True, blank=True)
    createdAt = models.DateTimeField(auto_now_add=True)
    updatedAt = models.DateTimeField(auto_now=True)

    class Meta:
        db_table = 'tblItemMenu'

    def __str__(self):
        return self.strLink
